File: algorithms/Palindrome/palindrome_no_v2.py
# Python program to find whether a number is a palindrome or not

# Function to reverse a number

def is_palindrome(n):
    return str(n) == str(n)[::-1]

num = int(input("Enter a number: "))
print("Entered number is a palindrome." if is_palindrome(num) else "Entered number is not a palindrome.")

'''
TEST CASES
INPUT
Enter a number:3445443
Entered number is palindrome.

OUTPUT
Enter a number:234
Entered number is not a palindrome.
'''

File: algorithms/Palindrome/palindrome_no_v2.c
/*
This program checks whether the enterd number is palindrome or not and generates output according to input provided.
*/

#include <stdio.h>
int main() {
    int n, temp, rev = 0;
    printf("Enter a number: ");
    scanf("%d", &n);
    for (temp = n; temp > 0; temp /= 10) {
        rev = rev * 10 + temp % 10;
    }
    printf("Entered number is %sa palindrome.\n", (n == rev) ? "" : "not ");
    return 0;
}

/*
  INPUT:
	n=12456
  OUTPUT:
  	Entered number is not palindrome.
  
  INPUT:
  	n=12821
  OUTPUT:
  	Entered number is palindrome.
*/